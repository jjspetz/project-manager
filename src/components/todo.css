.mainDisplay {
  width: 100%;
}
.flex-container {
  display: flex;
  align-content: stretch;
  align-items: stretch;
}

.card {
  position: relative;
  margin-top: 1%;
  margin-right: 20px;
  flex: 1 1 auto;
  text-align: center;
  min-height: 50vh;
}
.card:first-child {
  margin-left: 20px;
}
ol {
  list-style-position: inside;
  text-align: left;
}

li {
  margin: 5px 0;
  font-size: 20pt;
  width: 90%;
}
li:hover {
  background: #ccc;
}
.add {
  right: 4%;
  position: absolute;
  bottom: 3%;
}
.delete {
  margin: 1% 25%;
  width: 50%;
}
