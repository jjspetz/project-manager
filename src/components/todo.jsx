
// material ui
import AppBar from 'material-ui/AppBar';
import FlatButton from 'material-ui/FlatButton';
import IconButton from 'material-ui/IconButton';
import NavigationClose from 'material-ui/svg-icons/navigation/close';
import MenuItem from 'material-ui/MenuItem';
import Drawer from 'material-ui/Drawer';
// react and redux
import React, { Component } from 'react';
import {connect} from 'react-redux';
import {toggleSidebar} from '../actions';
import {bindActionCreators} from 'redux';
import Column from './column';
import './todo.css';
import database, {auth} from '../fire.js';
import DeleteArea from './deleteArea'

// add drop functions (basic script won't work)

class Todo extends Component {
  login () {
  auth()
    .then(function (user) {
      console.log(user)
    })
    .catch(function (e) {
      console.log(e);
    });
  }

  handleToggle = () => this.props.toggleSidebar(!this.props.openSidebar);

  handleClose = () => console.log('change later');

  render() {
    return (
      <div className='mainDisplay'>
        <AppBar
          title="TODO APP"
          onLeftIconButtonTouchTap={this.handleToggle}
          iconElementRight={<FlatButton label="Login" onTouchTap={(event)=> this.login()}/>}
        />
        <Drawer
          docked={false}
          width={250}
          open={this.props.openSidebar}
          onRequestChange={this.handleToggle}
        >
          <AppBar
            title="Projects"
            onLeftIconButtonTouchTap={this.handleToggle}
            iconElementLeft={<IconButton><NavigationClose /></IconButton>}
          />
          <MenuItem onTouchTap={this.handleClose}>+ New Project</MenuItem>
          <MenuItem onTouchTap={this.handleClose}>placeholder</MenuItem>
        </Drawer>
        <div className='flex-container'>
          <Column key='Uncompleted' title='Uncompleted'/>
          <Column key='In Progress' title='In Progress'/>
          <Column key='Completed' title='Completed'/>
        </div>
        <DeleteArea />
      </div>
    )
  };
}

function mapStateToProps(state) {
  return {
    openSidebar: state.openSidebar
  };
}

function mapDispatchToProps(dispatch) {
  return bindActionCreators(
    {
      toggleSidebar: toggleSidebar,
    }, dispatch
  )
}

export default connect(mapStateToProps, mapDispatchToProps)(Todo);
