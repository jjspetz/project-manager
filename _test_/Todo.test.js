import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Todo } from '../src/components/todo.jsx';

const wrapper = shallow(<Todo />);

describe('Todo component', () => {
  it('should render', () => {
    expect(wrapper).toBeTruthy();
  });
  // it('should open login on click', () => {
  //   expect(wrapper.find('DropdownLogin').exists()).toBe(false);
  //   wrapper.find('FlatButton').simulate('touchTap');
  //   expect(wrapper.find('DropdownLogin').exists()).toBe(false);
  // });
  it('should match snapshot', () => {
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
  });
});
