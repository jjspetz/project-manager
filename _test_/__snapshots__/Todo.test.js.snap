// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo component should match snapshot 1`] = `
<div
  className="mainDisplay"
>
  <AppBar
    iconElementRight={
      <FlatButton
        disabled={false}
        fullWidth={false}
        label="Login"
        labelPosition="after"
        labelStyle={Object {}}
        onKeyboardFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchStart={[Function]}
        onTouchTap={[Function]}
        primary={false}
        secondary={false}
      />
    }
    onLeftIconButtonTouchTap={[Function]}
    showMenuIconButton={true}
    title="Project Manager"
    zDepth={1}
  />
  <Connect(Sidebar) />
  <Connect(DropdownLogin) />
  <div
    className="flex-container"
  >
    <Connect(DropTarget(Column))
      key="Uncompleted"
      length={
        Object {
          "width": "62%",
        }
      }
      size="18pt"
      title="Uncompleted"
    />
    <Connect(DropTarget(Column))
      key="In Progress"
      length={
        Object {
          "width": "62%",
        }
      }
      size="18pt"
      title="In Progress"
    />
    <Connect(DropTarget(Column))
      key="Completed"
      length={
        Object {
          "width": "62%",
        }
      }
      size="18pt"
      title="Completed"
    />
  </div>
  <DropTarget(DeleteArea) />
  <div
    className="info"
  >
    <div>
      Project: 
      default
    </div>
    <div>
      version: 0.2.6
    </div>
    <div>
      Login to begin
    </div>
  </div>
</div>
`;
